vars/withOpenshiftServices.groovy

```diff
@@ -21,21 +21,39 @@ def cleanup(names) {
     }
 }
 
-def createOpenshiftResources(services, names) {
-    Map jobs = [:]
+def createOpenshiftResources(List<String> services, List<String> names) {
+    Map<String, Closure> jobs = [:]
     for (int i = 0; i < services.size(); i++) {
         String service = services[i]
         String name = names[i]
         jobs[name] = {
             openshiftCreateResource(getDeploymentConfigYaml(service, name))
             openshiftCreateResource(getServiceYaml(service, name))
             openshiftScale deploymentConfig: name,  replicaCount: 1, verifyReplicaCount: 1, waitTime: 600000
+            waitForServiceToBeReady service, name
         }
     }
     parallel jobs
 }
 
-String sanitizeObjectName(s) {
+def waitForServiceToBeReady(String service, String name) {
+    if (service == 'mongodb') {
+        writeFile(
+            file: 'checkMongo.sh',
+            text: "echo 'daves\ndaves()\n/' | curl -v telnet://${name}:27017/"
+        )
+        sh 'chmod +x checkMongo.sh'
+        timeout(5) {
```

Comment:
The '5' here is 5 minutes. I'm not sure what the timeout should be, or maybe it should be configurable? It shouldn't be hit in most cases hopefully.